src/Microsoft.VisualStudio.LanguageServices.Razor/Editor/DefaultVisualStudioDocumentTrackerFactory.cs

```diff
@@ -114,6 +114,25 @@ public override VisualStudioDocumentTracker GetTracker(ITextView textView)
             return tracker;
         }
 
+        public override VisualStudioDocumentTracker GetTracker(ITextBuffer textBuffer)
+        {
+            if (textBuffer == null)
+            {
+                throw new ArgumentNullException(nameof(textBuffer));
+            }
+
+            _foregroundDispatcher.AssertForegroundThread();
+
+            // A little bit of hardening here, to make sure our assumptions are correct.
+            DefaultVisualStudioDocumentTracker tracker;
+            if (!textBuffer.Properties.TryGetProperty(typeof(VisualStudioDocumentTracker), out tracker))
+            {
+                Debug.Fail("The document tracker should be initialized");
```

Comment:
This isn't right, what if the buffer isn't a razor buffer? This should just return null gracefully.

The ITextView version has the asserts because at that point we've already determined that it's a razor buffer

Reply:
Added a RazorBuffer check above and added tests.